### PR TITLE
Lead School to Lead Partner Part VI 

### PR DESCRIPTION
**Pull request by adamjcolvin**
### Context
[Trello](https://trello.com/c/vx9l5nmF)

This continues the refactoring of lead_school references throughout the app to lead_partner.

**Related PRs**
https://github.com/DFE-Digital/register-trainee-teachers/pull/4598 https://github.com/DFE-Digital/register-trainee-teachers/pull/4577 https://github.com/DFE-Digital/register-trainee-teachers/pull/4527 https://github.com/DFE-Digital/register-trainee-teachers/pull/4541 https://github.com/DFE-Digital/register-trainee-teachers/pull/4551

### Changes proposed in this pull request

* Renames the `lead_school` record_type on LeadPartner to just `school`. 
* Removes any remaining references to lead_school_id.  

### Important business

- [ ] Does this PR introduce any PII fields that need to be overwritten or deleted in db/scripts/sanitise.sql?
- [ ] Does this PR change the database schema? If so, have you updated the config/analytics.yml file and considered whether you need to send 'import_entity' events?
- [ ] Do we need to send any updates to DQT as part of the work in this PR?
- [ ] Does this PR need an ADR?

NB: Please notify the #twd_data_insights team and ask for a review if new fields are being added to analytics.yml

### CODE REVIEW

File: app/models/dttp/placement_assignment.rb [hunk 1]
```diff
@@ -39,10 +39,6 @@ def route_dttp_id
       response["_dfe_routeid_value"]
     end
 
-    def lead_school_id
```
Comment (defong) [inline app/models/dttp/placement_assignment.rb:42]: Not sure about this as the deals with `jsonb`

cc @stevehook can this class be deleted altogether given it's dttp related?

Comment (adamjcolvin) [reply app/models/dttp/placement_assignment.rb:42]: Yeah I did wonder if we could just delete the whole thing, but I wasn't sure. @stevehook what do you think?

Comment (adamjcolvin) [reply app/models/dttp/placement_assignment.rb:42]: Would we not have the historical data in a database backup somewhere? If they're not currently needed by the app, I would suggest we delete them. I could do that in another PR maybe. @tomtrentham what do you think?

Comment (tomtrentham) [reply app/models/dttp/placement_assignment.rb:42]: @adamjcolvin This feels separate to the current work to me. I think it's more closely related to the DTTP removal work that we've got in the tech debt backlog so I reckon we should leave this file now.

Comment (adamjcolvin) [reply app/models/dttp/placement_assignment.rb]: @tomtrentham no worries, I've undone this change for now then 👍

File: spec/factories/trainees.rb [hunk 2]
```diff
@@ -558,7 +558,7 @@
 
     trait :with_lead_partner do
       transient do
```
Comment (defong) [inline spec/factories/trainees.rb:560]: Don't think there was a need to wrap it in a `transient` block

File: spec/support/shared_examples/school_form_validations.rb [hunk 3]
```diff
@@ -4,11 +4,7 @@
   before { subject.valid? }
 
   let(:not_applicable) do
-    if school_id_key == "lead_school_id"
-      "lead_partner_not_applicable"
-    else
-      school_id_key.sub("id", "not_applicable")
-    end
+    school_id_key.sub("id", "not_applicable")
```
Comment (defong) [inline spec/support/shared_examples/school_form_validations.rb:7]: I don't think this shared example is shared anymore with lead `x`

**Review by defong — APPROVED** [review]
A few minor things otherwise I think it is all good

**Pull request merged**